static/js/utils/photometry.js

```diff
@@ -1,80 +1,203 @@
-      }<br>filter: ${element.filter}<br>limmag: ${
-        element.limiting_mag
-          ? element.limiting_mag.toFixed(4)
-          : element.limiting_mag
-      }<br>instrument: ${element.instrument_name}`,
-    };
-    data.showlegend = !names_already_seen.includes(name);
-    plot_data.push(data);
-    names_already_seen.push(name);
-  });
-  const layout = {
+      color: isDetection ? rgba(color, 0.3) : rgba(color, 0.1),
+      size: 10,
+      symbol: isDetection ? "circle" : "triangle-down",
+    },
+    hoverlabel: {
+      bgcolor: "white",
+      font: { size: 14 },
+      align: "left",
+    },
+    hovertemplate: "%{text}<extra></extra>",
+  };
+}
+
+function getLayout() {
+  return {
     autosize: true,
-    xaxis: { title: "Days ago", autorange: "reversed" },
-    yaxis: { title: "AB Mag", autorange: "reversed" },
+    xaxis: {
+      title: "Days Ago",
+      autorange: "reversed",
+      overlaying: "x",
+      ...baseLayout,
+    },
+    yaxis: {
+      title: "AB Mag",
+      autorange: "reversed",
+      ...baseLayout,
+    },
     margin: { l: 50, r: 50, b: 30, t: 30, pad: 1 },
+    shapes: [
+      {
+        type: "rect",
+        xref: "paper",
+        yref: "paper",
+        x0: 0,
+        y0: 0,
+        x1: 1,
+        y1: 1,
+        line: {
+          color: "black",
+          width: 1,
+        },
+      },
+    ],
+    legend: {
+      orientation: isMobile ? "h" : "v",
+      y: isMobile ? -0.3 : 1,
+      x: isMobile ? 0 : 1,
+    },
+    hovermode: "closest",
+  };
+}
+
+function getConfig() {
+  const resetLayout = () => {
+    Plotly.relayout(
+      document.getElementsByClassName("plotly")[0].parentElement,
+      getLayout(),
+    );
+  };
+  return {
+    responsive: true,
+    displaylogo: false,
+    showAxisDragHandles: false,
+    modeBarButtonsToRemove: [
+      "autoScale2d",
+      "resetScale2d",
+      "select2d",
+      "lasso2d",
+      "toggleSpikelines",
+      "hoverClosestCartesian",
+      "hoverCompareCartesian",
+    ],
+    modeBarButtonsToAdd: [
+      {
+        name: "Reset",
+        icon: Plotly.Icons.home,
+        click: () => {
+          resetLayout();
+        },
+      },
+    ],
   };
-  const config = { responsive: true };
+}
 
-  Plotly.newPlot(document.getElementById(div_id), plot_data, layout, config);
+function getGroupedPhotometry(photometry) {
+  return photometry.reduce((acc, point) => {
+    const key = `${point.instrument_name}/${point.filter}${
+      point.origin !== "None" ? `/${point.origin}` : ""
+    }`;
+    if (!acc[key]) {
+      acc[key] = [];
+    }
+    acc[key].push(point);
+    return acc;
+  }, {});
+}
+
+function adjustLegend() {
+  let isMobile = window.matchMedia("(max-width: 900px)").matches;
+  if (isMobile !== window.isMobile) {
+    window.isMobile = isMobile;
+    Plotly.relayout(
+      document.getElementsByClassName("plotly")[0].parentElement,
+      getLayout(),
+    );
+  }
+}
+
+window.addEventListener("resize", adjustLegend);
```

Comment:
I don't think we want an event listener added to a file that is important by the private page too, this will hurt performance for no good reason there as we don't use it. I think you could just have that listener created in your public page and pass it the callback